apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderInterceptor.java

```diff
@@ -96,6 +96,9 @@ private LogData transform(final AppenderSkeleton appender, LoggingEvent event) {
                         .build())
                 .setBody(LogDataBody.newBuilder().setType(LogDataBody.ContentCase.TEXT.name())
                                     .setText(TextLog.newBuilder().setText(transformLogText(appender, event)).build()).build());
+        if (ContextManager.getPrimaryEndpointName() != null) {
+            builder.setEndpoint(ContextManager.getPrimaryEndpointName());
```

Comment:
`getPrimaryEndpointName` doesn't have to be invoked twice. It is better `get name` and do null check?